rocketpy/stochastic/stochastic_flight.py

```diff
@@ -0,0 +1,145 @@
+            Whether or not the flight should terminate on apogee. This attribute
+            can not be randomized.
+        """
+        if terminate_on_apogee is not None:
+            assert isinstance(
+                terminate_on_apogee, bool
+            ), "`terminate_on_apogee` must be a boolean"
+        super().__init__(
+            flight,
+            rail_length=rail_length,
+            inclination=inclination,
+            heading=heading,
+        )
+
+        self.initial_solution = initial_solution
+        self.terminate_on_apogee = terminate_on_apogee
+
+    def _validate_initial_solution(self, initial_solution):
+        if initial_solution is not None:
+            if isinstance(initial_solution, (tuple, list)):
+                assert len(initial_solution) == 14, (
+                    "`initial_solution` must be a 14 element tuple, the "
+                    "elements are:\n t_initial, x_init, y_init, z_init, "
+                    "vx_init, vy_init, vz_init, e0_init, e1_init, e2_init, "
+                    "e3_init, w1Init, w2Init, w3Init"
+                )
+                assert all(
+                    isinstance(i, (int, float)) for i in initial_solution
+                ), "`initial_solution` must be a tuple of numbers"
+            else:
+                raise TypeError("`initial_solution` must be a tuple of numbers")
+
+    # TODO: these call dict_generator a lot of times unnecessarily
+    def _randomize_rail_length(self):
+        """Randomizes the rail length of the flight. Follows the standard input
+        format of Stochastic Models.
+        """
+        generated_dict = next(self.dict_generator())
+        return generated_dict["rail_length"]
+
+    def _randomize_inclination(self):
+        """Randomizes the inclination of the flight. Follows the standard input
+        format of Stochastic Models.
+        """
+        generated_dict = next(self.dict_generator())
+        return generated_dict["inclination"]
+
+    def _randomize_heading(self):
+        """Randomizes the heading of the flight. Follows the standard input
+        format of Stochastic Models.
+        """
+        generated_dict = next(self.dict_generator())
+        return generated_dict["heading"]
+
+    def create_object(self):
+        """Creates and returns a Flight object from the randomly generated input
+        arguments.
+
+        Returns
+        -------
+        flight : Flight
+            Flight object with the randomly generated input arguments.
+        """
+        generated_dict = next(self.dict_generator())
+        return Flight(
+            environment=self.object.env,
+            rail_length=self._randomize_rail_length(),
+            rocket=self.object.rocket,
+            inclination=generated_dict["inclination"],
+            heading=generated_dict["heading"],
+            initial_solution=self.initial_solution,
+            terminate_on_apogee=self.terminate_on_apogee,
+        )
```

Comment:
`randomize_inclination` and all these separated randomize methods are called in the Monte Carlo class during the simulation. The `create_object` method in this class is just here for consistency, and it might be useful for someone.

I have commented on this somewhere, but we could structure the `MonteCarlo` class without needing to initialize a `StochasticFlight` class. The inputs required for the simulation from the flight class should be specified when initializing the `MonteCarlo` object or when calling `.simulate()`

Reply:
I tried to play around with this, but could not come up with a good solution. Instead, I left it as a TODO, we can investigate deeper it the future.

[d4fd959](https://github.com/RocketPy-Team/RocketPy/pull/232/commits/d4fd959f408323c7514f77e60f8cef64aad1260e)